### DISCUSSION TITLE
Is it possible to ignore completion entries for specific LSP clients?

### DISCUSSION THREAD

**Discussion by megalithic** [hrsh7th/nvim-cmp#1643] [Q&A] [answered]
This is a bit edge-casey, but, I'm running 2 different lsp clients for the same language, to get certain benefits from both. I'd like to be able to ignore completion entries for 1 of those clients, but not the other. I thought I could use the `format` callback to just `return` if a specific `item.source.source.client.name` matches what I want to ignore/reject/skip, BUT, the `entry` module in nvim-cmp bomb expecting the return item to never be `nil`. 

Any suggestions of the proper way to ignore an item to form (or just ignore a specific lsp client source completely)?

Thanks!

**Comment by megalithic** [comment]
anyone facing similar needs? would love to get this working.

**Comment by fitrh** [comment, marked as answer]
I think you can disable the `completionProvider` capabilities by adding this to that specific server `on_attach`

```lua
    on_attach = function(client, bufnr)
      -- ...
      -- if you only use one on_attach function for every servers,
      -- you can also add something like `if client.name == '<NAME>' then`
      if client.server_capabilities.completionProvider then
        client.server_capabilities.completionProvider = false
      end
    end,
```

**Reply by megalithic** [reply]
BRILLIANT! 🤦

Thanks so much @fitrh!
